caffe2/core/context_base.cc

```diff
@@ -20,3 +20,12 @@ BaseStaticContext* get_static_context(DeviceType t) {
 }
 
 } // namespace caffe2
+
+namespace at {
```

Comment:
Err... this may pass the build, but it won't give you an ATen/core that is separately compileable (because if you ever try to use `ExtractDeviceOption` with only `ATen/core`, you won't find the symbol). Though, I guess this is probably harmless because we have no build configurations which use ATen/core but not caffe2 (it's a lot more severe for ATen/core versus ATen, since Caffe2 mobile doesn't use ATen).

Reply:
This is because we don't want to include caffe2 proto in ATen/core.

Reply:
I'll change to it to pure virtual..